server/pbenchinacan/deploy

```diff
@@ -1,26 +1,84 @@
 #!/bin/bash -e
 
-# Update the dashboard code
-podman unshare rm -r /srv/pbench/public_html/dashboard/
-podman unshare cp -r ${HOME}/git/pbench/dashboard/build /srv/pbench/public_html/dashboard
+#
+# This script deploys a Pbench Server instance from a container image.
+#
+# Steps performed:
+#   - The Pbench Dashboard application files are copied from the build directory
+#     into the SRV_PBENCH file system.  (If the build directory does not exist,
+#     an empty directory is created at the input and output locations, a warning
+#     is printed, and the Pbench Server is deployed without a Dashboard.)
+#   - If present in the current directory, the Nginx favicon file is copied into
+#     the appropriate location in the SRV_PBENCH file system.
+#   - The ownership of the files in the SRV_PBENCH file system will are set for
+#     access inside the container.
+#   - The Pbench Server container is started and run as a detached process.
+#
+# The key inputs are the following environment variables:
+#   - PB_DEPLOY_FILES:  the location (on the host) of the configuration files
+#                       for the Pbench Server and rsyslog.
+#   - SRV_PBENCH:  the location of the file system (on the host) which will be
+#                       mounted inside the container at /srv/pbench.
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
```

Comment:
The default values, for the most part, are drawn from the original `deploy` script which you created, @portante.  Are you happy with them as presented?
- The server image, and `/srv/pbench` are (I think) uncontroversial.
- I've added a pull policy of "newer" to override the Podman default of "missing"
- The default location of the Dashboard build assumes that the current directory is _in_ a Git checkout -- perhaps that is undesirable -- but you can specify either the location itself of the Git checkout, if the default (as currently spec'd) is unsuitable.
- The Nginx favicon file will be pulled from the Git checkout (by default) instead of coming from the current directory.  (Again, this can be redirected.)
- The biggest question is the default for the source of the deployment files -- do you like `${HOME}/Deploy`, or would you like to take this opportunity to change that?  (Once again, that's only the _default_ value -- you can use `PB_DEPLOY_FILES` to explicitly specify any directory you like.)

Reply:
The current default is fine for now.